# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../gettext/wchar-fix.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../gettext/wchar-fix.patch
# Copyright (C) 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
#
# Taken from: http://svn.exactcode.de/t2/trunk/package/base/gettext/wchar-fix.patch
#

diff -pruN gettext-0.17.orig//gettext-runtime/gnulib-lib/wchar.in.h gettext-0.17/gettext-runtime/gnulib-lib/wchar.in.h
--- gettext-0.17.orig//gettext-runtime/gnulib-lib/wchar.in.h	2010-02-17 15:36:18.822918689 +0100
+++ gettext-0.17/gettext-runtime/gnulib-lib/wchar.in.h	2010-02-17 15:38:37.686548691 +0100
@@ -26,6 +26,14 @@
  * the declaration of wcwidth().
  */
 
+#ifdef __need_mbstate_t
+/* Special invocation convention inside glibc header files.  */
+
+#@INCLUDE_NEXT@ @NEXT_WCHAR_H@
+
+#else
+/* Normal invocation convention.  */
+
 #ifndef _GL_WCHAR_H
 
 /* Tru64 with Desktop Toolkit C has a bug: <stdio.h> must be included before
@@ -80,3 +88,4 @@ extern int wcwidth (int /* actually wcha
 
 #endif /* _GL_WCHAR_H */
 #endif /* _GL_WCHAR_H */
+#endif
diff -pruN gettext-0.17.orig//gettext-tools/gnulib-lib/wchar.in.h gettext-0.17/gettext-tools/gnulib-lib/wchar.in.h
--- gettext-0.17.orig//gettext-tools/gnulib-lib/wchar.in.h	2010-02-17 15:36:21.302471149 +0100
+++ gettext-0.17/gettext-tools/gnulib-lib/wchar.in.h	2010-02-17 15:40:10.667180066 +0100
@@ -26,6 +26,14 @@
  * the declaration of wcwidth().
  */
 
+#ifdef __need_mbstate_t
+/* Special invocation convention inside glibc header files.  */
+
+#@INCLUDE_NEXT@ @NEXT_WCHAR_H@
+
+#else
+/* Normal invocation convention.  */
+
 #ifndef _GL_WCHAR_H
 
 /* Tru64 with Desktop Toolkit C has a bug: <stdio.h> must be included before
@@ -80,3 +88,4 @@ extern int wcwidth (int /* actually wcha
 
 #endif /* _GL_WCHAR_H */
 #endif /* _GL_WCHAR_H */
+#endif
diff -pruN gettext-0.17.orig//gettext-tools/libgettextpo/wchar.in.h gettext-0.17/gettext-tools/libgettextpo/wchar.in.h
--- gettext-0.17.orig//gettext-tools/libgettextpo/wchar.in.h	2010-02-17 15:36:19.870408142 +0100
+++ gettext-0.17/gettext-tools/libgettextpo/wchar.in.h	2010-02-17 15:40:48.206658667 +0100
@@ -26,6 +26,14 @@
  * the declaration of wcwidth().
  */
 
+#ifdef __need_mbstate_t
+/* Special invocation convention inside glibc header files.  */
+
+#@INCLUDE_NEXT@ @NEXT_WCHAR_H@
+
+#else
+/* Normal invocation convention.  */
+
 #ifndef _GL_WCHAR_H
 
 /* Tru64 with Desktop Toolkit C has a bug: <stdio.h> must be included before
@@ -80,3 +88,4 @@ extern int wcwidth (int /* actually wcha
 
 #endif /* _GL_WCHAR_H */
 #endif /* _GL_WCHAR_H */
+#endif
